### DISCUSSION TITLE
Does LSP4E support LSP over TCP?

### DISCUSSION THREAD

**Discussion by ghost** [eclipse/lsp4e#473] [Q&A]
I don't know where to ask so I posted the question here. Please pardon me if I posted in the wrong place. Thank you.

**Comment by jonahgraham** [comment]
The transport part is not implemented by core LSP4J, the interface to LSP4J is input and output streams you can connect to whatever you like, including a socket stream.

In addition, in the past people have contributed additional modules to provide a websocket connect in https://github.com/eclipse/lsp4j/tree/main/org.eclipse.lsp4j.websocket (or [jakarta](https://github.com/eclipse/lsp4j/tree/main/org.eclipse.lsp4j.websocket.jakarta) version)

Please see [Launch and connect with a LanguageClient](https://github.com/eclipse/lsp4j/blob/main/documentation/README.md#launch-and-connect-with-a-languageclient)

**Reply by ghost** [reply]
But this is LSP4E, isn't it? The language server used a custom tcp port. I wanted to know if it supports lsp over tcp and custom tcp port or not.

**Reply by jonahgraham** [reply]
Sorry, I saw you asked the same question as a discussion right after I answered the same question in https://github.com/eclipse/lsp4j/issues/696

Yes, this is in LSP4E. My answer is not relevant to LSP4E.

**Reply by ghost** [reply]
I changed `it` on the question's title to LSP4E. This is my carelessness that caused confusions. Sorry.
